src/Compilers/Core/Portable/Text/ChangedText.cs

```diff
@@ -370,10 +370,6 @@ private static void AddRange(List<TextChangeRange> list, TextChangeRange range)
                     list[list.Count - 1] = new TextChangeRange(new TextSpan(last.Span.Start, last.Span.Length + range.Span.Length), last.NewLength + range.NewLength);
                     return;
                 }
-                else
-                {
-                    Debug.Assert(range.Span.Start > last.Span.End);
```

Comment:
> Debug.Assert(range.Span.Start > last.Span.End); [](start = 19, length = 48)

I am not sure why is it the right thing to do to remove this assert completely. Could condition be adjusted instead to account for the specific situation? Explaining what is special about the situation would be good too.

Reply:
Looking at the original PR where this was added (https://github.com/dotnet/roslyn/pull/13446/):  it is unclear what @mattwar's intentions where.  My assumption was that we did not expect to encounter this case during normal usage, but it appears that https://github.com/dotnet/roslyn/issues/26305 is a trivial example of that not being the case.  Out of curiosity, if we launch VS with a debug build how often is this assert encountered?

Reply:
Would like to know this as well. Spoke with @heejaechang a bit and it sounds like there's probably an ordering assumption in the `List<TextChangeRange> list` being checked by this Assert.

Sounds like the way forward involves either making some change to the Merge method to fix the ordering of inputs that get provided or to make a change to `ChangedText.GetChangeRanges` to bail out and report the entire text as changed.